userspace/libsinsp/test/sinsp.ut.cpp

```diff
@@ -456,3 +456,47 @@ TEST_F(sinsp_with_test_input, execveat_invalid_path)
 		FAIL();
 	}
 }
+
+TEST_F(sinsp_with_test_input, creates_fd_generic)
+{
+	add_default_init_thread();
+
+	open_inspector();
+	sinsp_evt* evt = NULL;
+
+	add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_SIGNALFD_E, 3, -1, NULL, 0);
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_SIGNALFD_X, 1, 5);
+	ASSERT_EQ(get_field_as_string(evt, "fd.type"), "signalfd");
+	ASSERT_EQ(get_field_as_string(evt, "fd.typechar"), "s");
+	ASSERT_EQ(get_field_as_string(evt, "fd.num"), "5");
+
+	add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_TIMERFD_CREATE_E, 2, 0, 0);
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_TIMERFD_CREATE_X, 1, 6);
+	ASSERT_EQ(get_field_as_string(evt, "fd.type"), "timerfd");
+	ASSERT_EQ(get_field_as_string(evt, "fd.typechar"), "t");
+	ASSERT_EQ(get_field_as_string(evt, "fd.num"), "6");
+
+	add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_INOTIFY_INIT_E, 1, 0);
+	add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_INOTIFY_INIT_X, 1, 7);
```

Comment:
missing `evt =`?

Reply:
wait, how did CI pass without it? :thinking:

Reply:
Ah yes it's because of https://github.com/falcosecurity/libs/blob/000eb27573af74168510e4836330ff6958c3f355/userspace/libsinsp/sinsp.cpp#L1226. So basically the event is recycled by sinsp and so the pointer stays the same. Not a bug in the test framework, just so happens to work. Of course we can't rely on it so @FedeDP can you please just add the `evt =` so we're good? :smile:

Reply:
Fixed :)